packages/expo-updates/build/Updates.expo.js

```diff
@@ -50,6 +52,50 @@ export async function fetchUpdateAsync() {
         manifest: typeof result === 'string' ? JSON.parse(result) : result,
     };
 }
+// Legacy- Remove in SDK 39
+export async function clearUpdateCacheExperimentalAsync(sdkVersion) {
+    const errors = [];
+    if (Platform.OS !== 'android') {
+        errors.push('This method is only supported on Android.');
+        return { success: false, errors };
+    }
+    if (Constants.manifest && FileSystem.documentDirectory) {
+        const sdkBundlesPath = FileSystem.documentDirectory + sdkVersion ?? Constants.manifest.sdkVersion;
+        const sdkBundleFiles = await FileSystem.readDirectoryAsync(sdkBundlesPath);
+        sdkBundleFiles.forEach(async (filename) => {
+            let fullpath = sdkBundlesPath + '/' + filename;
+            // In java, we use `getPath`, which decodes, so we need to double-encode these values
+            fullpath = fullpath.replace('%40', '%2540').replace('%2F', '%252F');
+            const bundleUrlStringHashcode = hashCode(Constants.manifest.bundleUrl);
+            const isCurrentlyRunningBundle = filename.includes(bundleUrlStringHashcode);
+            if (!isCurrentlyRunningBundle) {
+                try {
+                    await FileSystem.deleteAsync(fullpath);
+                }
+                catch (e) {
+                    errors.push(e);
+                }
+            }
+        });
+        if (!errors.length) {
```

Comment:
This should `map` with an `await Promise.all(...)` instead of `forEach` otherwise `errors` will (almost?) always be empty.

Reply:
yeah that's a good catch, writing a fix for that now